From: Lars-Peter Clausen <lars@metafoo.de>
Date: Wed, 16 Nov 2011 10:13:36 +0100
Subject: I2C: Add helper macro for i2c_driver boilerplate

commit 7c92784a546d2945b6d6973a30f7134be78eb7a4 upstream.

This patch introduces the module_i2c_driver macro which is a convenience macro
for I2C driver modules similar to module_platform_driver. It is intended to be
used by drivers which init/exit section does nothing but register/unregister
the I2C driver. By using this macro it is possible to eliminate a few lines of
boilerplate code per I2C driver.

Signed-off-by: Lars-Peter Clausen <lars@metafoo.de>
Acked-by: Grant Likely <grant.likely@secretlab.ca>
Acked-by: Jonathan Cameron <jic23@cam.ac.uk>
Acked-by: Wolfram Sang <w.sang@pengutronix.de>
Acked-by: Jean Delvare <khali@linux-fr.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 include/linux/i2c.h |   13 +++++++++++++
 1 file changed, 13 insertions(+)

Index: linux-3.2.46/include/linux/i2c.h
===================================================================
--- linux-3.2.46.orig/include/linux/i2c.h	2013-07-26 18:53:46.000000000 +0000
+++ linux-3.2.46/include/linux/i2c.h	2013-07-26 19:25:46.000000000 +0000
@@ -482,6 +482,19 @@
 {
 	return adap->nr;
 }
+
+/**
+ * module_i2c_driver() - Helper macro for registering a I2C driver
+ * @__i2c_driver: i2c_driver struct
+ *
+ * Helper macro for I2C drivers which do not do anything special in module
+ * init/exit. This eliminates a lot of boilerplate. Each module may only
+ * use this macro once, and calling it replaces module_init() and module_exit()
+ */
+#define module_i2c_driver(__i2c_driver) \
+	module_driver(__i2c_driver, i2c_add_driver, \
+			i2c_del_driver)
+
 #endif /* I2C */
 #endif /* __KERNEL__ */
 
